src/custom/state/lists/updater.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect } from 'react'
+import { unstable_batchedUpdates as batchedUpdates } from 'react-dom'
+import { useDispatch } from 'react-redux'
+
+import { useActiveWeb3React } from '@src/hooks'
+
+import { DEFAULT_ACTIVE_LIST_URLS_BY_NETWORK, DEFAULT_LIST_OF_LISTS_BY_NETWORK } from 'constants/lists'
+import { initialiseTokenLists } from './actions'
+import { ListsState, NEW_LIST_STATE } from './reducer'
+import { buildKey, buildListState } from './helpers'
+
+export default function NetworkListUpdater(): null {
+  const { chainId } = useActiveWeb3React()
+  // const chainIdRef = useRef(chainId)
+
+  const dispatch = useDispatch()
+
+  // Loading//Saving network specific lists from//into storage
+  useEffect(() => {
+    if (chainId) {
+      let refreshedTokenState: ListsState
+      // build identifier for storage token list e.g Rinkeby list identifier = "_4"
+      const newChainKey = buildKey(chainId)
+      // Get the localStorage token list saved under identifer key
+      // e.g "redux_localstorage_simple_lists_4"
+      const storageListJSON = localStorage.getItem(newChainKey)
+
+      // storageListJSON && localStorage.setItem(newChainKey, storageListJSON)
+      if (storageListJSON) {
```

Comment:
that logic is almost 1:1 what was in the commented out reducer case. There's functionally no difference. 

The reason it is here is to avoid changing every action and subsequently every file using that action to add a chainId. Better to do it there in one place imo